src/enzo/ActiveParticle.C

```diff
@@ -0,0 +1,411 @@
+    my_type->InitializeParticleType();
+
+    EnabledActiveParticles[EnabledActiveParticlesCount++] = my_type;
+    int this_id = my_type->Enable();
+    /* Note that this ID is only unique for each individual instantiation of
+       Enzo.  The next time you start it, these numbers might be different.
+       This is why they aren't output to a file!
+    */
+    assert(my_type->GetEnabledParticleID() == EnabledActiveParticlesCount-1);
+    return;
+}
+
+int ActiveParticleType::ReadDataset(int ndims, hsize_t *dims, const char *name,
+                    hid_t group, hid_t data_type, void *read_to)
+{
+  hid_t file_dsp_id;
+  hid_t dset_id;
+  hid_t h5_status;
+  herr_t      h5_error = -1;
+  int i, j, k, dim;
+  /* get data into temporary array */
+
+  file_dsp_id = H5Screate_simple((Eint32) ndims, dims, NULL);
+  if( file_dsp_id == h5_error ){ENZO_FAIL("Error creating file dataspace");}
+
+  dset_id =  H5Dopen(group, name);
+  if( dset_id == h5_error )ENZO_VFAIL("Error opening %s", name)
+
+  h5_status = H5Dread(dset_id, data_type, H5S_ALL, H5S_ALL, H5P_DEFAULT, (VOIDP) read_to);
+  if( dset_id == h5_error )ENZO_VFAIL("Error reading %s", name)
+
+  h5_status = H5Sclose(file_dsp_id);
+  if( dset_id == h5_error )ENZO_VFAIL("Error closing dataspace %s", name)
+
+  h5_status = H5Dclose(dset_id);
+  if( dset_id == h5_error )ENZO_VFAIL("Error closing %s", name)
+
+  return SUCCESS;
+}
+
+int ActiveParticleType::WriteDataset(int ndims, hsize_t *dims, const char *name, hid_t group,
+				     hid_t data_type, void *data)
+{
+
+  hid_t file_dsp_id;
+  hid_t dset_id;
+  hid_t h5_status;
+  herr_t h5_error = -1;
+
+  file_dsp_id = H5Screate_simple((Eint32) ndims, dims, NULL);
+  if( file_dsp_id == h5_error )
+    ENZO_VFAIL("Error creating dataspace for %s", name)
+
+  dset_id =  H5Dcreate(group, name, data_type, file_dsp_id, H5P_DEFAULT);
+  if( dset_id == h5_error )
+    ENZO_VFAIL("Error creating dataset %s", name)
+
+  h5_status = H5Dwrite(dset_id, data_type, H5S_ALL, H5S_ALL, H5P_DEFAULT,
+		       (VOIDP) data);
+  if( h5_status == h5_error )
+    ENZO_VFAIL("Error writing dataset %s", name)
+
+  h5_status = H5Sclose(file_dsp_id);
+  if( h5_status == h5_error )
+    ENZO_VFAIL("Error closing dataspace %s", name)
+
+  h5_status = H5Dclose(dset_id);
+  if( h5_status == h5_error )
+    ENZO_VFAIL("Error closing dataset %s", name)
+
+  return SUCCESS;
+}
+
+
+
+void ActiveParticleType::ConstructData(grid *_grid,
+            ActiveParticleFormationDataFlags &flags,
+            ActiveParticleFormationData &data) {
+
+  /* We have a number of items that can be required; we now attempt to
+     generate them.
+
+     This uses code from the old grid::StarParticleHandler routine. */
+
+
+  /* initialize */
+
+  int dim, i, j, k, index, size, field, GhostZones = NumberOfGhostZones;
+  int DensNum, GENum, TENum, Vel1Num, Vel2Num, Vel3Num, B1Num, B2Num, B3Num,H2INum, H2IINum;
+  const double m_h = 1.673e-24;
```

Comment:
This is already defined as "mass_h" in phys_const.def

Reply:
It doesn't appear to be used anyway so I deleted it.